spec/acceptance/api/internal/services_api_spec.rb

```diff
@@ -81,21 +83,39 @@
       # The returned data should not contain *some_param* attribute
       expect(svc.to_hash).to eq service.merge(user_registration_required: true)
     end
+
+    context 'with an service that has no state' do
+      let (:state) { nil }
+
+      example_request 'Trying to create the service' do
+        expect(status).to eq 201
+        expect(response_json['status']).to eq 'created'
+
+        svc = ThreeScale::Backend::Service.load_by_id('1002')
+        svc_hash = svc.to_hash
+        expect(svc_hash).to have_key(:state)
+        expect(svc_hash.delete(:state)).to be(:active)
```

Comment:
unfortunately we are coupling here with the implementation (or part of it with `to_hash`) - I know this is already there for other cases, but we should not continue that bad trend. Better to either test that the JSON response has `state: "active"` (if the created object is included in the response, which I believe it might not be at all) or test that the loaded service object has `active?` set to true.